guides/source/active_record_basics.md

```diff
@@ -71,88 +67,105 @@ Convention over Configuration in Active Record
 
+* **Primary keys** - By default, Active Record will use an integer column named
+  `id` as the table's primary key (`bigint` for PostgreSQL and MySQL, `integer`
+  for SQLite). When using [Active Record
+  Migrations](active_record_migrations.html) to create your tables, this column
+  will be automatically created.
 * **Foreign keys** - These fields should be named following the pattern
   `singularized_table_name_id` (e.g., `item_id`, `order_id`). These are the
   fields that Active Record will look for when you create associations between
   your models.
-* **Primary keys** - By default, Active Record will use an integer column named
-  `id` as the table's primary key (`bigint` for PostgreSQL and MySQL, `integer`
-  for SQLite). When using [Active Record Migrations](active_record_migrations.html)
-  to create your tables, this column will be automatically created.
 
-There are also some optional column names that will add additional features
-to Active Record instances:
+There are also some optional column names that will add additional features to
+Active Record instances:
 
 * `created_at` - Automatically gets set to the current date and time when the
   record is first created.
 * `updated_at` - Automatically gets set to the current date and time whenever
   the record is created or updated.
 * `lock_version` - Adds [optimistic
-  locking](https://api.rubyonrails.org/classes/ActiveRecord/Locking.html) to
-  a model.
+  locking](https://api.rubyonrails.org/classes/ActiveRecord/Locking.html) to a
+  model.
 * `type` - Specifies that the model uses [Single Table
   Inheritance](https://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Single+table+inheritance).
-* `(association_name)_type` - Stores the type for
-  [polymorphic associations](association_basics.html#polymorphic-associations).
+* `(association_name)_type` - Stores the type for [polymorphic
+  associations](association_basics.html#polymorphic-associations).
 * `(table_name)_count` - Used to cache the number of belonging objects on
-  associations. For example, a `comments_count` column in an `Article` class that
-  has many instances of `Comment` will cache the number of existent comments
-  for each article.
+  associations. For example, a `comments_count` column in an `Article` class
+  that has many instances of `Comment` will cache the number of existent
+  comments for each article.
 
-NOTE: While these column names are optional, they are in fact reserved by Active Record. Steer clear of reserved keywords unless you want the extra functionality. For example, `type` is a reserved keyword used to designate a table using Single Table Inheritance (STI). If you are not using STI, try an analogous keyword like "context", that may still accurately describe the data you are modeling.
+NOTE: While these column names are optional, they are reserved by Active Record.
+Steer clear of reserved keywords when naming your table's columns. For example,
```

Comment:
I'm wondering if there is a link to somewhere in the guides where have a list of reserved words that would be helpful in this context, if not maybe we can add a task on the board to create one?

Reply:
Good point, and there was a discussion a few years ago about this on the forum: https://discuss.rubyonrails.org/t/reserved-model-names/74559/19 

Added it as [a task](https://3.basecamp.com/3076981/buckets/35498807/todos/7121438912) for now and will let @carlosantoniodasilva weigh in here to see if we should do this (or if it exists somewhere that he is aware of).

Reply:
I don't think there's a defined list anywhere, it's more about the columns AR uses (which are listed here), but there's things like defining a column named `save` or other AR methods could cause you trouble. 

Rails has some [checks for internal methods](https://github.com/rails/rails/blob/c402ec78724a305d2e22777986aa6f2068d5a0ae/activerecord/lib/active_record/attribute_methods.rb#L30) and it might raise out loud, but it's may not catch everything.

I think it'd be somewhat difficult to keep an up-to-date list of methods, but the columns we have listed here are a pretty good starting point.

FWIW, there's a [similar thing with Action Controller in that guide](https://edgeguides.rubyonrails.org/action_controller_overview.html#methods-and-actions)
![Screenshot 2024-02-26 at 14 37 57](https://github.com/bhumi1102/rails/assets/26328/d2d919b1-6f68-459c-ab4d-e4c234bbb68a)